spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -117,14 +144,16 @@ private DispatchQueryResponse handleIndexQuery(
                 .build()
                 .toString(),
             tags,
-            indexDetails.getAutoRefresh());
+            indexDetails.getAutoRefresh(),
+            dataSourceMetadata.getResultIndex());
     String jobId = emrServerlessClient.startJobRun(startJobRequest);
-    return new DispatchQueryResponse(jobId, false);
+    return new DispatchQueryResponse(jobId, false, dataSourceMetadata.getResultIndex());
```

Comment:
what if resultIndex is null?

Reply:
if it is null, we will use default result index as before.